src/raster/rasterband.rs

```diff
@@ -556,6 +562,41 @@ impl<'a> RasterBand<'a> {
         Array2::from_shape_vec((size.1, size.0), data).map_err(Into::into)
     }
 
+    #[cfg(feature = "ndarray")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "array")))]
+    /// Write a [`Array2<T>`] from a [`Dataset`] block, where `T` implements [`GdalType`].
+    ///
+    /// # Arguments
```

Comment:
Can we add a reference to the method for getting the block layout (so one can know how many blocks there are in each dimension)?

Reply:
An example would be helpful here.

Reply:
Added an example and a note to `block_size` and `actual_block_size`.